src/warp.js

```diff
@@ -44,155 +49,301 @@ function getLanguage() {
+    addClass(homeHrefImage, 'img');
+    homeHref.appendChild(homeHrefImage);
+    homeHrefListElement.appendChild(homeHref);
+    return homeHrefListElement;
+}
+
+function createMenu(categories) {
+    var menuContainer = document.createElement('div');
+    addClass(menuContainer, 'warp-menu-column-menu');
+    addClass(menuContainer, 'menu-container-hide');
+
+    var menu = document.createElement('div');
+    addClass(menu, 'warp-menu-shift-container');
+    menuContainer.appendChild(menu);
+
+    var overlay = document.createElement('div');
+    addClass(overlay, 'warp-menu-gradient-overlay');
+    menuContainer.appendChild(overlay);
+
+    var list = document.createElement('ul');
+    addClass(list, 'warp-menu-category-list');
+    list.id = 'warp-menu-category-list';
+    menu.appendChild(list);
+
+    var homeHrefElement = createHomeHrefWithImage();
+    list.appendChild(homeHrefElement);
 
     for (var c = 0; c < categories.length; c++) {
-        var category = categories[c];
+        var currentCategory = categories[c];
 
-        if (category.Title.toUpperCase() === "INFORMATION") {
-            informationEntries = category.Entries;
+        if (currentCategory.Title.toUpperCase() === "INFORMATION") {
+            informationEntries = currentCategory.Entries;
         } else {
-            var id = getCategoryKey(category);
-            createMenuEntry(id, category.Entries, category.Title, nav);
+            var title = currentCategory.Title;
+            if (isTranslateable(title)) {
+                title = getLocalizedString(title);
+            }
+            var id = getCategoryKey(currentCategory);
+            createMenuEntry(id, currentCategory.Entries, title, list);
         }
     }
 
     // fixed about page - entry
     informationEntries.push({
-        DisplayName: ABOUT_CLOUDOGU_TOKEN,
+        DisplayName: getLocalizedString("aboutCloudoguToken"),
         Href: createLink("/info/index.html")
     });
+    createMenuEntry("warpc.info", informationEntries, "Information", list);
+
+    addLogoutMenuEntry(list);
+
+    window.addEventListener('resize', setCorrectColumnCount);
+
+    return menuContainer;
+}
 
-    createMenuEntry("warpc.info", informationEntries, "Information", nav);
+function initWarpMenu(categories) {
+    var warpMenuContainer = document.createElement('div');
+    addClass(warpMenuContainer, 'warp-menu-container');
+    warpMenuContainer.id = 'warp-menu-container';
 
-    var div = document.createElement('div');
-    addClass(div, 'warpbtn');
-    var btn = document.createElement('a');
-    addClass(btn, 'warpbtn-link');
+    var tooltipColumn = createTooltip();
+    var toggleResult = createToggleButton();
+    var menuContainer = createMenu(categories);
 
     function toggleNav() {
-        toggleClass(div, 'warpbtn-open');
-        toggleClass(nav, 'warpmenu-open');
-        toggleClass(body, 'warpmenu-push-toleft');
+        if (hasClass(warpMenuContainer, 'collapsing')) return;
```

Comment:
For better readability the `return` statement should be surrounded by curly brackets.